test/functional/discovered_hosts_controller_test.rb

```diff
@@ -73,6 +73,13 @@ def test_edit_form_attributes
     assert_not_nil host.cpu_count
   end
 
+  def test_edit_form_build_mode_enabled
+    host = Host::Discovered.import_host(@facts)
```

Comment:
Oh one more thing, can you use this helper instead now:

    host = discover_host_from_facts(@facts)